src/components/CourseFilter/CourseFilter.tsx

```diff
@@ -0,0 +1,192 @@
+      setSelectedTags(updatedTags);
+    } else {
+      updatedTags = [...selectedTags, tagName];
+      setSelectedTags(updatedTags);
+    }
+    if (tagName === '') {
+      setSelectedTags([]);
+    }
+    const tagsAsaString = updatedTags.filter((tag) => tag).join(',');
+    router.push(pathname + '?' + createQueryString('courseTag', tagsAsaString));
+  };
+
+  const handleDateChange = async (range: [any, any]) => {
+    const [startDate, endDate] = range;
+    setStartDate(startDate);
+    setEndDate(endDate);
+    if (startDate === null && endDate === null) {
+      router.push(pathname + '?' + createQueryString('courseDates', ''));
+    } else {
+      router.push(
+        pathname +
+          '?' +
+          createQueryString('courseDates', startDate + '-' + endDate)
+      );
+    }
+  };
+
+  const handleClearSearch = async () => {
+    await handleNameChange(null);
+    await handleTagChange('');
+    await handleDateChange([null, null]);
+    router.replace(pathname);
+  };
+
+  const customStyles = {
+    width: '250px',
+    height: '55px',
+    fontSize: '16px',
+  };
+
+  return (
+    <>
+      <Typography variant="h4">Search courses</Typography>
+      <Box
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          padding: '16px',
+          backgroundColor: palette.white.main,
+        }}
+      >
+        <Button
+          variant="contained"
+          onClick={() => handleClearSearch()}
+          sx={{
+            width: '200px',
+            marginRight: '50px',
+            backgroundColor: palette.surface.main,
+          }}
+        >
+          Clear search
+        </Button>
+        <div>
+          <Autocomplete
+            value={courseName || null}
+            clearOnEscape
+            disablePortal
+            data-testid="search-autocomplete"
+            id="combo-box"
+            options={initialCourses.map((course) => course.name)}
+            renderOption={(props, option) => {
+              return (
+                <li {...props} key={option}>
+                  {option}
+                </li>
+              );
+            }}
+            sx={{ width: '250px', marginRight: '50px' }}
+            renderInput={(value) => (
+              <TextField {...value} label="Search by a name" />
+            )}
+            onChange={(event, value) => {
+              handleNameChange(value);
+            }}
+          />
+        </div>
+        <DatePicker
+          fixedHeight
+          placeholderText="  Search by dates"
+          minDate={new Date()}
+          selected={startDate}
+          onChange={handleDateChange}
+          startDate={startDate}
+          endDate={endDate}
+          selectsRange
+          showWeekNumbers
+          isClearable
+          highlightDates={initialCourses.map((course) => course.startDate)}
+          customInput={<input style={customStyles} />}
+        />
+      </Box>
+      <FormControl>
+        <ButtonGroup>
```

Comment:
The button group might not be optimal for this. The tags overflow out of the viewport pretty fast. On mobile the tag search is most likely fully unfunctional. The UI becomes generally also very tricky to handle.

See the CourseForm implementation for tags. Hiding the tags in a similiar Menu would be sustainable. It would be responsive and the UI be manageable. Otherwise the tag search relies on there being just limited amount of tags.